swift/ql/lib/codeql/swift/generated/ParentChild.qll

```diff
@@ -0,0 +1,2182 @@
+      nArgument =
+        n + 1 + max(int i | i = 0 or exists(e.(SubscriptExpr).getImmediateArgument(i)) | i) and
+      (
+        none()
+        or
+        result = e.(SubscriptExpr).getImmediateArgument(index - n) and
+        partialAccessor = "Argument(" + (index - n).toString() + ")"
+      )
+    )
+    or
+    exists(int n, int nExpr, int nCase |
+      n = 0 + getMaximumChildrenIndexLabeledStmt(e) and
+      nExpr = n + 1 and
+      nCase = nExpr + 1 + max(int i | i = 0 or exists(e.(SwitchStmt).getImmediateCase(i)) | i) and
+      (
+        none()
+        or
+        index = n and result = e.(SwitchStmt).getImmediateExpr() and partialAccessor = "Expr()"
+        or
+        result = e.(SwitchStmt).getImmediateCase(index - nExpr) and
+        partialAccessor = "Case(" + (index - nExpr).toString() + ")"
+      )
+    )
+    or
+    exists(int n, int nBody |
+      n = 0 + getMaximumChildrenIndexLabeledConditionalStmt(e) and
+      nBody = n + 1 and
+      (
+        none()
+        or
+        index = n and result = e.(GuardStmt).getImmediateBody() and partialAccessor = "Body()"
+      )
+    )
+    or
+    exists(int n, int nThen, int nElse |
+      n = 0 + getMaximumChildrenIndexLabeledConditionalStmt(e) and
+      nThen = n + 1 and
+      nElse = nThen + 1 and
+      (
+        none()
+        or
+        index = n and result = e.(IfStmt).getImmediateThen() and partialAccessor = "Then()"
+        or
+        index = nThen and result = e.(IfStmt).getImmediateElse() and partialAccessor = "Else()"
+      )
+    )
+    or
+    exists(int n, int nParam |
+      n =
+        0 + getMaximumChildrenIndexAbstractStorageDecl(e) + getMaximumChildrenIndexGenericContext(e) and
+      nParam = n + 1 + max(int i | i = 0 or exists(e.(SubscriptDecl).getImmediateParam(i)) | i) and
+      (
+        none()
+        or
+        result = e.(SubscriptDecl).getImmediateParam(index - n) and
+        partialAccessor = "Param(" + (index - n).toString() + ")"
+      )
+    )
+    or
+    exists(int n, int nBody |
+      n = 0 + getMaximumChildrenIndexLabeledConditionalStmt(e) and
+      nBody = n + 1 and
+      (
+        none()
+        or
+        index = n and result = e.(WhileStmt).getImmediateBody() and partialAccessor = "Body()"
+      )
+    )
+  }
+}
+
+/**
+ * Gets the "immediate" parent of `e`. "Immediate" means not taking into account node resolution: for example
+ * if `e` has conversions, `getImmediateParent(e)` will give the bottom conversion in the hidden AST.
+ */
+Element getImmediateParent(Element e) {
+  // `unique` is used here to tell the optimizer that there is in fact only one result
+  // this is tested by the `library-tests/parent/no_double_parents.ql` test
+  result = unique(Element x | e = Impl::getImmediateChild(x, _, _) | x)
+}
+
+/**
+ * Gets the immediate child indexed at `index`. Indexes are not guaranteed to be contiguous, but are guaranteed to be distinct. `accessor` is bound the the method giving the given child.
```

Comment:
QL doesn't have methods 😄. Could we say "member-predicate name" instead?

Reply:
good point. Is it a member predicate call then? It's not just for example `getElement`, it's `getElement(3)`

Reply:
Yeah, I think that sounds good 👌